packages/odyssey-react-mui/src/DataTable/DataTable.tsx

```diff
@@ -0,0 +1,488 @@
+    positionActionsColumn:
+      "last" as MRT_TableOptions<MRT_RowData>["positionActionsColumn"],
+    renderRowActions: ({ row }) => renderRowActions({ row }),
+
+    // Row selection
+    enableRowSelection: hasRowSelection,
+
+    // Sorting
+    enableSorting: hasSorting,
+    onSortingChange: setColumnSorting,
+    muiTableHeadCellProps: ({ column: currentColumn }) => ({
+      className: columnSorting.find(
+        (sortedColumn) => sortedColumn.id === currentColumn.id
+      )
+        ? "isSorted"
+        : "isUnsorted",
+    }),
+
+    // Virtualization
+    enableRowVirtualization:
+      paginationType === "loadMore" || pagination.pageSize > 50,
+    rowVirtualizerInstanceRef:
+      useRef<MRT_Virtualizer<HTMLDivElement, HTMLTableRowElement>>(null),
+    rowVirtualizerOptions: {
+      overscan: 4,
+    },
+
+    // Filters
+    renderTopToolbar: () => (
+      <DataFilters
+        onChangeSearch={hasSearch ? setSearch : undefined}
+        onChangeFilters={setFilters}
+        hasSearchSubmitButton={hasSearchSubmitButton}
+        searchDelayTime={searchDelayTime}
+        filters={hasFilters ? dataTableFilters : undefined}
+        additionalActions={
+          <DataTableSettings
+            hasChangeableDensity={hasChangeableDensity}
+            rowDensity={rowDensity}
+            setRowDensity={setRowDensity}
+            hasColumnVisibility={hasColumnVisibility}
+            columns={columns}
+            columnVisibility={columnVisibility}
+            setColumnVisibility={setColumnVisibility}
+          />
+        }
+      />
+    ),
+
+    // Pagination
+    renderBottomToolbar: hasPagination
+      ? () => (
+          <DataTablePagination
+            paginationType={paginationType}
+            currentNumberOfResults={data.length}
+            currentPage={pagination.pageIndex}
+            isPreviousButtonDisabled={pagination.pageIndex <= 1}
+            isNextButtonDisabled={false} // TODO: Add logic for disabling next/load more button
+            onClickPrevious={() =>
+              setPagination({
+                pageIndex: pagination.pageIndex - 1,
+                pageSize: pagination.pageSize,
+              })
+            }
+            onClickNext={() => {
+              if (paginationType === "loadMore") {
+                setPagination({
+                  pageSize: pagination.pageSize,
+                  pageIndex: pagination.pageSize + resultsPerPage,
+                });
+              } else {
+                setPagination({
+                  pageSize: pagination.pageSize,
+                  pageIndex: pagination.pageIndex + 1,
+                });
+              }
+            }}
+          />
+        )
+      : undefined,
+  });
+
+  // Effects
+  useEffect(() => {
+    onChangeRowSelection?.(dataTable.getState().rowSelection);
+  }, [dataTable.getState().rowSelection]);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const incomingData = await onChangeQuery?.({
```

Comment:
You call await on the function passed here (which is great!) but the type on L137 does not indicate that this could be an async function. This will be necessary if anyone needs to do async operations to fetch more data (e.g. a loadMore-type pagination). I think this is as easy as changing the type from `=> MRT_TableOptions<MRT_RowData>["data"]` to something like `=> MRT_TableOptions<MRT_RowData>["data"] | Promise<MRT_TableOptions<MRT_RowData>["data"]>` but I could be wrong about the exact syntax.